relay-metrics/src/bucket.rs

```diff
@@ -556,10 +565,105 @@ impl From<MetricValue> for BucketValue {
     }
 }
 
-/// Error returned when parsing or serializing a [`Bucket`].
-#[derive(Debug, Error)]
-#[error("failed to parse metric bucket")]
-pub struct ParseBucketError(#[source] serde_json::Error);
+/// Parses a list of counter values separated by colons and sums them up.
+fn parse_counter(string: &str) -> Option<CounterType> {
+    let mut sum = CounterType::default();
+    for component in string.split(':') {
+        sum += component.parse::<CounterType>().ok()?;
+    }
+    Some(sum)
+}
+
+/// Parses a distribution from a list of floating point values separated by colons.
+fn parse_distribution(string: &str) -> Option<DistributionValue> {
+    let mut dist = DistributionValue::default();
+    for component in string.split(':') {
+        dist.insert(component.parse().ok()?);
+    }
+    Some(dist)
+}
+
+/// Parses a set of hashed numeric values.
+fn parse_set(string: &str) -> Option<SetValue> {
+    let mut set = SetValue::default();
+    for component in string.split(':') {
+        set.insert(component.parse().ok()?);
+    }
+    Some(set)
+}
+
+/// Parses a gauge from a value.
+///
+/// The gauge can either be given as a single floating point value, or as a list of exactly five
+/// values in the order of [`GaugeValue`] fields.
+fn parse_gauge(string: &str) -> Option<GaugeValue> {
+    let mut components = string.split(':');
+
+    let first = components.next()?.parse().ok()?;
+    Some(if let Some(min) = components.next() {
+        GaugeValue {
+            max: first,
+            min: min.parse().ok()?,
```

Comment:
Kinda weird to have `max` before `min`. Maybe we should flip them in both the protocol and the `GaugeValue` type?